### DISCUSSION TITLE
How to make a React component a module

### DISCUSSION THREAD

**Discussion by Hmfazlt** [GrapesJS/grapesjs#6315] [General]
![image](https://github.com/user-attachments/assets/6647b190-24f1-43c2-aa8b-0b6c9214a1f1)

**Comment by Hmfazlt** [comment]
https://codesandbox.io/s/grapesjs-react-components-n6sff
This example solution usually does not work

**Comment by danstarns** [comment]
@Hmfazlt what do you mean by _usually*_?

**Comment by Hmfazlt** [comment]
![image](https://github.com/user-attachments/assets/39b01bd9-6452-49cb-8e28-0f2a315b3925)
![image](https://github.com/user-attachments/assets/2abb592f-8bd2-47fa-a6c0-063238dd03e9)
The components of the UI framework will lose their functionality,Actually, it should be the following effect:
![image](https://github.com/user-attachments/assets/956ad494-c02b-489c-9222-ca65e4a88a62)

**Comment by Hmfazlt** [comment]
Another question is: after using React components as Block; How to export to other websites for use? If exported as HTML, the components will not display properly.
![image](https://github.com/user-attachments/assets/62122963-77de-47a4-bbbc-d968992c2ba2)

**Reply by artf** [reply]
The example you are referring to is mainly to show how to render React components inside the canvas but for the code export is up to you figure it out.
